src/effects/backends/builtin/beatgrindeffect.cpp

```diff
@@ -129,64 +111,48 @@ void BeatGrindEffect::processChannel(
-        delay_samples = gs.delay_buf.size();
+    int loopSamples = loopFrames * bufferParameters.channelCount();
+    VERIFY_OR_DEBUG_ASSERT(loopSamples <= gs.loop.size()) {
+        loopSamples = gs.loop.size();
     }
 
-    RampingValue<CSAMPLE_GAIN> send(send_current, gs.prev_send,
-                                    bufferParameters.framesPerBuffer());
-    // Feedback the delay buffer and then add the new input.
-
-    RampingValue<CSAMPLE_GAIN> feedback(feedback_current, gs.prev_feedback,
-                                        bufferParameters.framesPerBuffer());
-    
     for (unsigned int frame = 0;
             frame < bufferParameters.samplesPerBuffer();
             frame += bufferParameters.channelCount()) {
-        CSAMPLE_GAIN send_ramped = send.getNext();
-        CSAMPLE_GAIN feedback_ramped = feedback.getNext();    
-
+  
         for (int channel = 0; channel < bufferParameters.channelCount(); channel++) {
+            // record loop
             if (gs.isRecording) {
-                if (gs.write_position >= gs.delay_buf.size()) {
+                if (gs.writeSamplePos >= gs.loop.size()) {
                     gs.isRecording = false;
                 } else {
-                    gs.delay_buf[gs.write_position] = pInput[frame + channel];
-                    gs.write_position += 1;
+                    gs.loop[gs.writeSamplePos] = pInput[frame + channel];
+                    gs.writeSamplePos += 1;
                 }
             }
-            if (gs.write_position >= delay_samples) {
-                if (gs.read_position >= delay_samples) {
-                    gs.read_position = 0;
+            // if there is nothing to output yet passtrough dry signal??
+            CSAMPLE wetValue = pInput[frame + channel];
+            // play loop
+            if (gs.writeSamplePos >= loopSamples) {
+                if (gs.readSamplePos >= loopSamples) {
+                    gs.readSamplePos = 0;
                 }
-                gs.bufferedSample[channel] = gs.delay_buf[gs.read_position];
-                gs.read_position += 1;
+                wetValue = gs.loop[gs.readSamplePos];
+                gs.readSamplePos += 1;
             }
-            pOutput[frame + channel] = SampleUtil::clampSample(
-                pInput[frame + channel] * send_ramped +
-                gs.bufferedSample[channel] * feedback_ramped);
+            pOutput[frame + channel] = 
+                pInput[frame + channel] * (1 - mixValue) +
+                wetValue * mixValue;
         }
     }
-    // The ramping of the send parameter handles ramping when enabling, so
-    // this effect must handle ramping to dry when disabling itself (instead
-    // of being handled by EngineEffect::process).
-    if (enableState == EffectEnableState::Disabling) {
-        SampleUtil::applyRampingGain(pOutput, 1.0, 0.0, bufferParameters.samplesPerBuffer());
-        gs.delay_buf.clear();
-        gs.prev_send = 0;
```

Comment:
I think we need to the same ramping off when disabling as Echo does. Similar to Echo, ramping the Mix parameter will ramp the effect on, so mark `setEffectRampsFromDry(true)` on the manifest.